l3build-stdmain.lua

```diff
@@ -28,17 +28,15 @@ local exit   = os.exit
 local insert = table.insert
 
 -- List all modules
-function listmodules()
+local function listmodules()
   local modules = { }
   local exclmodules = exclmodules or { }
-  for entry in lfs.dir(".") do
-    if entry ~= "." and entry ~= ".." then
-      local attr = lfs.attributes(entry)
-      assert(type(attr) == "table")
-      if attr.mode == "directory" then
-        if not exclmodules[entry] then
-          insert(modules, entry)
-        end
+  for entry in tree(".", "*/build.lua") do -- TODO: support "**/..."
+    local module = dirname(entry.cwd)
+    if #module > 2 then -- this is not the current directory
+      module = module:sub(3) -- remove the leading "./"
+      if not exclmodules[module] then
+        insert(modules, module)
```

Comment:
I don't follow - aren't you risking having unpredictable nesting levels?

Reply:
No, "*/..." only looks at the first level contrary to "**/..." that looks at arbitrary depth

Reply:
Ah, right, yes. 

So wider question is why not just add to `exclmodules` as in my suggestion - minimal change and all that.